### ISSUE TITLE
No support for SSL/TLS connections with HHVM

### ISSUE THREAD

**Issue by clue** [reactphp/socket-client#27]
HHVM does not (currently) implement the required `stream_socket_enable_crypto()` function (https://github.com/facebook/hhvm/issues/1647).

Trying to establish a SSL/TLS connection via the `SecureConnector` results in a fatal error:

```
Fatal error: Call to undefined function stream_socket_enable_crypto() […]
```

Opening this ticket as a reminder and for reference in related issues.

**Comment by clue** [comment]
Until HHVM adds support we may want to look into providing a workaround to at least fail gracefully.

Checking for the existence of the function and rejecting the promise would probably be the least invasive way that does not involve a BC break.

**Comment by flip111** [comment]
This has been implemented since hhvm 3.8 https://github.com/facebook/hhvm/blob/HHVM-3.8/hphp/runtime/ext/stream/ext_stream.cpp#L690
